api/metadata.go

```diff
@@ -0,0 +1,85 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/equinor/vds-slice/internal/core"
+)
+
+// MetadataGet godoc
+// @Summary  Return volumetric metadata about the VDS
+// @description.markdown metadata
+// @Tags     metadata
+// @Param    query  query  string  True  "Urlencoded/escaped MetadataRequest"
+// @Produce  json
+// @Success  200 {object} core.Metadata
+// @Failure  400 {object} ErrorResponse "Request is invalid"
+// @Failure  500 {object} ErrorResponse "openvds failed to process the request"
+// @Router   /metadata  [get]
+func (e *Endpoint) MetadataGet(ctx *gin.Context) {
+	var request MetadataRequest
+	err := parseGetRequest(ctx, &request)
+	if abortOnError(ctx, err) {
+		return
+	}
+
+	e.metadata(ctx, request)
+}
+
+// MetadataPost godoc
+// @Summary  Return volumetric metadata about the VDS
+// @description.markdown metadata
+// @Tags     metadata
+// @Param    body  body  MetadataRequest  True  "Request parameters"
+// @Produce  json
+// @Success  200 {object} core.Metadata
+// @Failure  400 {object} ErrorResponse "Request is invalid"
+// @Failure  500 {object} ErrorResponse "openvds failed to process the request"
+// @Router   /metadata  [post]
+func (e *Endpoint) MetadataPost(ctx *gin.Context) {
+	var request MetadataRequest
+	err := parsePostRequest(ctx, &request)
+	if abortOnError(ctx, err) {
+		return
+	}
+
+	e.metadata(ctx, request)
+}
+
+func (e *Endpoint) metadata(ctx *gin.Context, request MetadataRequest) {
```

Comment:
I did consider it since there are some potential changes that would lead to both of these functions changing. In the end I decided to move it since the purpose of the common file is to keep code that is shared among multiple controllers.

I am still considering some more refactoring of these files, and unifying metadata and the other requests a bit more is on the list.